### PR TITLE
XHR: Split up xmlhttprequest-timeout-worker-synconworker.html

### PR DESCRIPTION
**Pull request by zcorpan**
Part of #11571.

### CODE REVIEW

**Review by jdm — APPROVED** [review]

**Comment by zcorpan** [comment]
I think I'll add some more commits here for similar xhr tests.

**Comment by zcorpan** [comment]
OK please take a look.

**Comment by zcorpan** [comment]
Chrome and Firefox runs timed out, will need to force-merge.

**Review by annevk — APPROVED** [review]
Since @jdm is happy go ahead. It'd be kinda nice to rewrite these tests as they're hard to follow, but that's too much work.
